src/main/java/picard/arrays/MergePedIntoVcf.java

```diff
@@ -0,0 +1,260 @@
+
+        final VariantContextWriter writer = new VariantContextWriterBuilder()
+                .setOutputFile(output)
+                .setReferenceDictionary(dict)
+                .setOptions(VariantContextWriterBuilder.DEFAULT_OPTIONS)
+                .build();
+
+        writer.writeHeader(vcfHeader);
+        while (variants.hasNext()) {
+            final VariantContext context = variants.next();
+
+            final VariantContextBuilder builder = new VariantContextBuilder(context);
+            if (zCallThresholds.containsKey(context.getID())) {
+                final String[] zThresh = zCallThresholds.get(context.getID());
+                builder.attribute(InfiniumVcfFields.ZTHRESH_X, zThresh[0]);
+                builder.attribute(InfiniumVcfFields.ZTHRESH_Y, zThresh[1]);
+            }
+            final Genotype originalGenotype = context.getGenotype(0);
+            final Map<String, Object> newAttributes = originalGenotype.getExtendedAttributes();
+            final VCFEncoder vcfEncoder = new VCFEncoder(vcfHeader, false, false);
+            final Map<Allele, String> alleleMap = vcfEncoder.buildAlleleStrings(context);
+
+            final String zCallAlleles = zCallPedFile.getAlleles(context.getID());
+            if (zCallAlleles == null) {
+                throw new PicardException("No zCall alleles found for snp " + context.getID());
+            }
+            final List<Allele> zCallPedFileAlleles = buildNewAllelesFromZCall(zCallAlleles, context.getAttributes());
+            newAttributes.put(InfiniumVcfFields.GTA, alleleMap.get(originalGenotype.getAllele(0)) + "/" + alleleMap.get(originalGenotype.getAllele(1)));
+            newAttributes.put(InfiniumVcfFields.GTZ, alleleMap.get(zCallPedFileAlleles.get(0)) + "/" + alleleMap.get(zCallPedFileAlleles.get(1)));
+
+            final Genotype newGenotype = GenotypeBuilder.create(originalGenotype.getSampleName(), zCallPedFileAlleles,
+                    newAttributes);
+            builder.genotypes(newGenotype);
+            logger.record("0", 0);
+            // AC, AF, and AN are recalculated here
+            VariantContextUtils.calculateChromosomeCounts(builder, false);
+            final VariantContext newContext = builder.make();
+            if (!zCallPedFileAlleles.equals(originalGenotype.getAlleles())) {
+                newContext.getCommonInfo().addFilter(InfiniumVcfFields.ZCALL_DIFF);
+            }
+            writer.add(newContext);
+        }
+
+        writer.close();
+    }
+
+    private List<Allele> buildNewAllelesFromZCall(final String zCallPedFileAlleles,
+                                                  final Map<String, Object> newAttributes) {
+        final char allele1 = zCallPedFileAlleles.charAt(0);
+        final char allele2 = zCallPedFileAlleles.charAt(1);
+        final List<Allele> newAlleles = new ArrayList<>();
+        final String alleleA = String.valueOf(newAttributes.get("ALLELE_A"));
```

Comment:
should `"ALLELE_A"` and `"ALLELE_B"` be defined as constants somewhere?

Reply:
Whoops, they already are